manifests/repo.pp

```diff
@@ -12,15 +12,19 @@
 #
 # @param gpgkey
 #   The GPG key to use
+#
+# @param baseurl
+#   An optional base URL to be used for yumrepo, instead of the default
 class candlepin::repo (
-  String $version,
+  String $version = '4.9',
```

Comment:
I'm leaning to keep it as a required string. If we had candlepin specific repositories (like with Pulpcore) it'd be different, but while we keep tracking Katello versions I'd keep it empty:
```suggestion
  String $version,
```

For for acceptance tests we already default to nightly:
https://github.com/theforeman/puppet-candlepin/blob/c21d2192f98a5bbb8aa65e88702644709692f326/spec/setup_acceptance_node.pp#L8

You can use `BEAKER_FACTER_candlepin_version=4.9` if you quickly want to override this.

Reply:
The situation that felt weird was when you are specifying a baseurl, and are forced to specify a version but then the version is not used at all because you specified a baseurl. It provides a contradictory configuration.

Reply:
If that's the case, I'd rather make it `Variant[Undef, Enum['nightly'], Pattern[/^\d+\.\d+$/]] $version`. Then in the body itself assert it's not undef when `$baseurl` is not provided.
```puppet
unless $baseurl {
  assert_type(NotUndef, $version)
}